// @aws-sdk/client-timestream-write/dist-es/commands/DeleteTableCommand.js
import { __extends } from "tslib";
import { getEndpointDiscoveryPlugin } from "@aws-sdk/middleware-endpoint-discovery";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteTableRequest } from "../models/models_0";
import { deserializeAws_json1_0DeleteTableCommand, serializeAws_json1_0DeleteTableCommand, } from "../protocols/Aws_json1_0";
var DeleteTableCommand = (function (_super) {
    __extends(DeleteTableCommand, _super);
    function DeleteTableCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteTableCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointDiscoveryPlugin(configuration, { clientStack: clientStack, options: options, isDiscoveredEndpointRequired: true }));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TimestreamWriteClient";
        var commandName = "DeleteTableCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteTableRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteTableCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_0DeleteTableCommand(input, context);
    };
    DeleteTableCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_0DeleteTableCommand(output, context);
    };
    return DeleteTableCommand;
}($Command));
export { DeleteTableCommand };
