// @aws-sdk/client-timestream-write/dist-es/commands/DescribeDatabaseCommand.js
import { __extends } from "tslib";
import { getEndpointDiscoveryPlugin } from "@aws-sdk/middleware-endpoint-discovery";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeDatabaseRequest, DescribeDatabaseResponse } from "../models/models_0";
import { deserializeAws_json1_0DescribeDatabaseCommand, serializeAws_json1_0DescribeDatabaseCommand, } from "../protocols/Aws_json1_0";
var DescribeDatabaseCommand = (function (_super) {
    __extends(DescribeDatabaseCommand, _super);
    function DescribeDatabaseCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDatabaseCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointDiscoveryPlugin(configuration, { clientStack: clientStack, options: options, isDiscoveredEndpointRequired: true }));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TimestreamWriteClient";
        var commandName = "DescribeDatabaseCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDatabaseRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeDatabaseResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDatabaseCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_0DescribeDatabaseCommand(input, context);
    };
    DescribeDatabaseCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_0DescribeDatabaseCommand(output, context);
    };
    return DescribeDatabaseCommand;
}($Command));
export { DescribeDatabaseCommand };
