// @aws-sdk/client-timestream-write/dist-es/commands/ListTablesCommand.js
import { __extends } from "tslib";
import { getEndpointDiscoveryPlugin } from "@aws-sdk/middleware-endpoint-discovery";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTablesRequest, ListTablesResponse } from "../models/models_0";
import { deserializeAws_json1_0ListTablesCommand, serializeAws_json1_0ListTablesCommand, } from "../protocols/Aws_json1_0";
var ListTablesCommand = (function (_super) {
    __extends(ListTablesCommand, _super);
    function ListTablesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTablesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointDiscoveryPlugin(configuration, { clientStack: clientStack, options: options, isDiscoveredEndpointRequired: true }));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TimestreamWriteClient";
        var commandName = "ListTablesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTablesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListTablesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTablesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_0ListTablesCommand(input, context);
    };
    ListTablesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_0ListTablesCommand(output, context);
    };
    return ListTablesCommand;
}($Command));
export { ListTablesCommand };
