// @aws-sdk/client-timestream-write/dist-es/commands/UpdateDatabaseCommand.js
import { __extends } from "tslib";
import { getEndpointDiscoveryPlugin } from "@aws-sdk/middleware-endpoint-discovery";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateDatabaseRequest, UpdateDatabaseResponse } from "../models/models_0";
import { deserializeAws_json1_0UpdateDatabaseCommand, serializeAws_json1_0UpdateDatabaseCommand, } from "../protocols/Aws_json1_0";
var UpdateDatabaseCommand = (function (_super) {
    __extends(UpdateDatabaseCommand, _super);
    function UpdateDatabaseCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateDatabaseCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointDiscoveryPlugin(configuration, { clientStack: clientStack, options: options, isDiscoveredEndpointRequired: true }));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TimestreamWriteClient";
        var commandName = "UpdateDatabaseCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateDatabaseRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateDatabaseResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateDatabaseCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_0UpdateDatabaseCommand(input, context);
    };
    UpdateDatabaseCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_0UpdateDatabaseCommand(output, context);
    };
    return UpdateDatabaseCommand;
}($Command));
export { UpdateDatabaseCommand };
